nautobot/dcim/api/views.py

```diff
@@ -593,14 +593,37 @@ class PowerOutletViewSet(PathEndpointMixin, CustomFieldModelViewSet):
     brief_prefetch_fields = ["device"]
 
 
-class InterfaceViewSet(PathEndpointMixin, CustomFieldModelViewSet):
+@extend_schema_view(
+    bulk_update=extend_schema(
+        responses={"200": serializers.InterfaceSerializerVersion13(many=True)}, versions=["1.2", "1.3"]
+    ),
+    bulk_partial_update=extend_schema(
+        responses={"200": serializers.InterfaceSerializerVersion13(many=True)}, versions=["1.2", "1.3"]
+    ),
+    create=extend_schema(responses={"201": serializers.InterfaceSerializerVersion13}, versions=["1.2", "1.3"]),
+    list=extend_schema(responses={"200": serializers.InterfaceSerializerVersion13(many=True)}, versions=["1.2", "1.3"]),
+    partial_update=extend_schema(responses={"200": serializers.InterfaceSerializerVersion13}, versions=["1.2", "1.3"]),
+    retrieve=extend_schema(responses={"200": serializers.InterfaceSerializerVersion13}, versions=["1.2", "1.3"]),
+    update=extend_schema(responses={"200": serializers.InterfaceSerializerVersion13}, versions=["1.2", "1.3"]),
+)
+class InterfaceViewSet(PathEndpointMixin, CustomFieldModelViewSet, StatusViewSetMixin):
     queryset = Interface.objects.prefetch_related(
-        "device", "_path__destination", "cable", "_cable_peer", "ip_addresses", "tags"
+        "device", "status", "_path__destination", "cable", "_cable_peer", "ip_addresses", "tags"
     )
     serializer_class = serializers.InterfaceSerializer
     filterset_class = filters.InterfaceFilterSet
     brief_prefetch_fields = ["device"]
 
+    def get_serializer_class(self):
+        serializer_choices = (
+            SerializerVersions(versions=["1.2", "1.3"], serializer=serializers.InterfaceSerializerVersion13),
+        )
+        return versioned_serializer_selector(
+            obj=self,
+            serializer_choices=serializer_choices,
+            current_serializer=super().get_serializer_class(),
+        )
```

Comment:
Maybe just me, but this API seems rather unintuitive.

- Why does `SerializerVersions` (plural) take a single serializer as an input? Would calling it something like "SerializerForAPIVersions(versions, serializer)` be clearer?
- How does `current_serializer` relate to the above versions? Is this the serializer that gets used if none of the serializer_choices match the current API version, or something else?

I personally would have expected something like the below pseudocode instead:

```
serializer_choices = {
    "1.2": serializers.InterfaceSerializerVersion12,
    "1.3": serializers.InterfaceSerializerVersion12,
    "latest": serializers.InterfaceSerializer
}
return versioned_serializer_selector(serializer_choices, self)
```

Reply:
> ```
> serializer_choices = {
>     "1.2": serializers.InterfaceSerializerVersion12,
>     "1.3": serializers.InterfaceSerializerVersion12,
>     "latest": serializers.InterfaceSerializer
> }
> return versioned_serializer_selector(serializer_choices, self)
> ```

I don't think this approach is great, just imagine having several APIs using the same serializer, too much repetition IMO
```
 serializer_choices = {
    "1.2": serializers.InterfaceSerializerVersion12,
    "1.3": serializers.InterfaceSerializerVersion12,
    "1.4": serializers.InterfaceSerializerVersion12,
    "1.5": serializers.InterfaceSerializerVersion12,
    "1.6": serializers.InterfaceSerializerVersion12,
    "1.7": serializers.InterfaceSerializerVersion12,
    "latest": serializers.InterfaceSerializer
 }
```